### DISCUSSION TITLE
installing kubernetes

### DISCUSSION THREAD

**Discussion by Stargate-project** [authelia/authelia#5743] [Q&A] [answered]
Should I follow the installation on the official site (https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/), or is there another, quicker way to get Kubernetes up and running?

I know its a bit different on aws EC2 instances, should I [Installing Kubernetes with kOps](https://kubernetes.io/docs/setup/production-environment/tools/kops/)

or can I use [Bootstrapping clusters with kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/)

**Comment by james-d-elliott** [comment, marked as answer]
There are so many ways it's probably impossible to list all of them. I'd say k3d is one of the easier ways, kubeadm is a fairly commonly supported / standard way however.

**Comment by Stargate-project** [comment]
I wasn't sure what you guys use, and I never heard of K3d, but I will give it a try. I want something that is easy to upgrade in the future.
Can I use K3d together with authelia, will there be any conflicts?

Thanks for the info.

**Reply by james-d-elliott** [reply]
We perform our k8s integration testing with k3d so it should be fine. I personally use kubeadm with weave-net and traefik, looking to move to kubeadm with cilium and traefik. I also externally terminate TLS and load balance to my cluster via haproxy.
